packages/process/src/node/raw-process.ts

```diff
@@ -27,9 +27,11 @@ export interface RawProcessFactory {
 @injectable()
 export class RawProcess extends Process {
 
+    readonly input: stream.Writable;
     readonly output: stream.Readable;
     readonly errorOutput: stream.Readable;
-    protected process: child.ChildProcess;
+    // XXX: Do we have to make this public? How to attach additional listeners to the underlying process then?
+    readonly process: child.ChildProcess;
```

Comment:
I did not want to shadow the process to allow direct API access to it since I wasn't sure yet what was needed.

I think it may be better to leave it public in case a user wants to do something non trivial with it and we don't want to wrap the whole API in raw-process.

Not sure what you mean by attaching additional listeners ?

Reply:
> be better to leave it public

I agree. Otherwise, I could not listen on errors for instance [here](https://github.com/theia-ide/theia/blob/33c7342b1b34d9924a74af0e802fca9080cf8a2f/packages/languages/src/node/language-server-contribution.ts#L73).  One can check why [`stderr`](https://github.com/Tyriar/node-pty/issues/71) is not available on the `ITerminal`. I hope this answers your last question about the listeners too.